### DISCUSSION TITLE
sidecar-container: failed, failed to start container.

### DISCUSSION THREAD

**Discussion by lucacampanile3** [testground/testground#1100] [Q&A] [answered]
#### Version information:
Version latest

#### Description:
Hi guys, i'm trying to run ipfs testcase https://github.com/ipfs/test-plans/tree/master/bitswap-tuning  and when i run the test with command `testground run single --plan ipfs/bitswap-tuning --testcase transfer  --builder docker:go --runner local:docker --instances 10` the testground daemon create various container such as influxdb,redis,grafana,goproxy and at least testground-sidecar  for the testing but when it creates testground-sidecar the `engine run` stops and print an error :  

> healthcheck fixes failed; aborting:
> Checks:
> - local-outputs-dir: ok; directory exists.
> - control-network: ok; network exists.
> - local-grafana: ok; container state: running
> - local-redis: ok; container state: running
> - local-influxdb: ok; container state: running
> - sidecar-container: failed; container state: created
> Fixes:
> - local-outputs-dir: unnecessary; 
> - control-network: unnecessary; 
> - local-grafana: unnecessary; 
> - local-redis: unnecessary; 
> - local-influxdb: unnecessary; 
> - sidecar-container: failed; failed to start container. 

I've looked around the docs  and i checked that i can run manually the side-car with `$ testground sidecar --runner local:docker` but still not going
I'm trying to run this command with MacOS and Ubuntu OS but still nothing. In particular if i use this command in MacOS it will print 
> the sidecar only supports linux, not darwin

Maybe the sidecar is not supported in unix system?
I'm not sure if this is a bug or my own misunderstanding of how testground underlying tools work
How can i solve this?

Thanks a lot for the answer

**Comment by raulk** [comment, marked as answer]
Hey! Thanks for trying out Testground. Did you run `make install` from the root of the repo? That will build the sidecar image, which appears to be missing in your machine.

**Comment by nonsense** [comment]
@LucaCampanile to add to Raul's answer - the `sidecar` does not work on macOS, but we compile it inside a Docker image, so you can still use Testground on macOS with Docker, via the `local:docker` runner (which I see you are trying to do based on your command).

**Comment by lucacampanile3** [comment]
> Hey! Thanks for trying out Testground. Did you run `make install` from the root of the repo? That will build the sidecar image, which appears to be missing in your machine.

Hi thanks a lot for your reply, i don't know why i forgot to run `make install` from the root of the repo, maybe i use this command in another repo and that's why don't go sidecar container , thanks a lot for resolve my problem 
Now the test is running but i have another question.
After i runned the test all the data from the testground goes into influxDB, as  i can see  testground doesn't setup automatically the data source for InfluxDB in Grafana, so i setup manually to link InfluxDB in Grafana. But for see the results i need to query the data inside influxDB but how i can see the data stored inside influxDB?
I'm trying to run inside influxDB container query for seeing all the data but nothing happen. So my question is how i can see the data of testground stored in InfluxDB?
Thanks a lot for your answer, 
I apologize if I have published another problem, if it is not lawful to do it I can also close this problem and open another one
Thank you in advance

**Reply by raulk** [reply]
@nonsense Grafana should have some autocomplete/explore features, right?
@LucaCampanile personally I like running Chronograf as it shows the schema in a nice manner.
I run it with:

```
$ docker run --network testground-control -p 8888:8888 chronograf --influxdb-url=http://testground-influxdb:8086
```

**Reply by raulk** [reply]
Some useful screenshots.

![image](https://user-images.githubusercontent.com/1101242/87813537-f9fd6800-c859-11ea-9cd8-3ce8d4639119.png)

![image](https://user-images.githubusercontent.com/1101242/87813460-df2af380-c859-11ea-90d4-9b431595b4fa.png)

**Comment by nonsense** [comment]
> After i runned the test all the data from the testground goes into influxDB, as i can see testground doesn't setup automatically the data source for InfluxDB in Grafana, so i setup manually to link InfluxDB in Grafana. But for see the results i need to query the data inside influxDB but how i can see the data stored inside influxDB?

That's right - Testground doesn't setup automatically the data source for InfluxDB in Grafana - see my answer at https://github.com/testground/testground/issues/1098#issuecomment-656325684

Ideally in the future we will have a more automated setup with dashboards provided by the teams, so that anyone can reproduce results from test plans.

I see that you are running the `bitswap-tuning` test plans - these were some of the very first test plans created in Testground (AFAIK by @dirkmc) and results have been analysed in custom scripts from the `outputs`, not from the data that is pushed to InfluxDB. Usually we `collect` all `outputs` a test plan generates and process them in Jupyter. However there is no automated way right now part of Testground to automatically reproduce this process.

**Comment by nonsense** [comment]
@raulk 

> @nonsense Grafana should have some autocomplete/explore features, right?

Yes, there is an `explore` function in Grafana, but ultimately it is hard to use that, unless you are the person who has implemented the test plan.

The whole "data visualisation and insights gathering" flow when analysing test runs (be it `outputs` or `diagnostics` or `events`), is not automated so that folks can randomly pick a test plan, run it and explore what has happened in the tests as well as it results in an easy-to-use way.
